posthog/temporal/batch_exports/bigquery_batch_export.py

```diff
@@ -629,53 +660,61 @@ async def insert_into_bigquery_activity(inputs: BigQueryInsertInputs) -> Records
-            ):
-                records_completed = await run_consumer_loop(
-                    queue=queue,
-                    consumer_cls=BigQueryConsumer,
-                    producer_task=producer_task,
-                    heartbeater=heartbeater,
-                    heartbeat_details=details,
-                    data_interval_end=data_interval_end,
-                    data_interval_start=data_interval_start,
-                    schema=record_batch_schema,
-                    writer_format=WriterFormat.PARQUET,
-                    max_bytes=settings.BATCH_EXPORT_BIGQUERY_UPLOAD_CHUNK_SIZE_BYTES,
-                    json_columns=(),
-                    bigquery_client=bq_client,
-                    bigquery_table=bigquery_stage_table,
+                    table_id=stage_table_name if has_query_permissions else inputs.table_id,
                     table_schema=stage_schema,
-                    writer_file_kwargs={"compression": "zstd"},
-                    multiple_files=True,
-                )
-
-                merge_key = (
-                    bigquery.SchemaField("team_id", "INT64"),
-                    bigquery.SchemaField("distinct_id", "STRING"),
-                )
-                await bq_client.amerge_tables(
-                    final_table=bigquery_table,
-                    stage_table=bigquery_stage_table,
-                    mutable=True if model_name == "persons" else False,
-                    merge_key=merge_key,
-                    stage_fields_cast_to_json=json_columns,
-                )
+                    create=has_query_permissions,
+                    delete=has_query_permissions,
+                ) as bigquery_stage_table:
+                    records_completed = await run_consumer_loop(
+                        queue=queue,
+                        consumer_cls=BigQueryConsumer,
+                        producer_task=producer_task,
+                        heartbeater=heartbeater,
+                        heartbeat_details=details,
+                        data_interval_end=data_interval_end,
+                        data_interval_start=data_interval_start,
+                        schema=record_batch_schema,
+                        writer_format=WriterFormat.PARQUET if has_query_permissions else WriterFormat.JSONL,
+                        max_bytes=settings.BATCH_EXPORT_BIGQUERY_UPLOAD_CHUNK_SIZE_BYTES,
+                        json_columns=() if has_query_permissions else json_columns,
+                        bigquery_client=bq_client,
+                        bigquery_table=bigquery_stage_table if has_query_permissions else bigquery_table,
+                        table_schema=stage_schema if has_query_permissions else schema,
+                        writer_file_kwargs={"compression": "zstd"} if has_query_permissions else {},
+                        multiple_files=True,
+                    )
+
+                    if has_query_permissions:
```

Comment:
again a small nitpick but think it would improve readability if this variable was named something like `perform_merge` (as without context it's not immediately obvious why you need to merge tables if you have query permissions and you'd probably have to scroll up to see why. Again, not a big deal but is probably simpler)